core/src/main/scala/org/apache/spark/deploy/history/ApplicationCache.scala

```diff
@@ -223,7 +224,10 @@ private[history] class ApplicationCache(
     }
   }
 
-  def invalidate(key: CacheKey): Unit = appCache.invalidate(key)
+  def invalidate(key: CacheKey): Unit = appCache.asMap().computeIfPresent(key, (key, value) => {
```

Comment:
> @LuciferYang This can be accomplished by using cache.asMap().computeIfPresent instead of invalidate. Then you may callback manually within the atomic operation and return null to remove the entry.

@ben-manes Do you mean that?

Reply:
yes. I'd pass `RemovalCause.EXPLICIT`, just to avoid null

Reply:
ok